ngraph/test/preprocess.cpp

```diff
@@ -107,6 +107,137 @@ TEST(pre_post_process, tensor_element_type_and_scale) {
+                       .preprocess(PreProcessSteps().convert_color(ColorFormat::RGB)))
+            .build(f);
+
+    EXPECT_EQ(f->get_parameters().size(), 2);
+    EXPECT_EQ(f->get_parameters()[0]->get_element_type(), element::f32);
+    EXPECT_EQ(f->get_parameters()[1]->get_element_type(), element::f32);
+    EXPECT_EQ(f->get_parameters()[0]->get_partial_shape(), (PartialShape{5, 2, 2, 1}));
+    EXPECT_EQ(f->get_parameters()[1]->get_partial_shape(), (PartialShape{5, 1, 1, 2}));
+}
+
+TEST(pre_post_process, convert_color_nv12_bgr_2_planes_u8_lvalue) {
+    auto f = create_simple_function(element::u8, Shape{1, 2, 2, 3});
+    auto input_tensor_info = InputTensorInfo();
+    input_tensor_info.set_color_format(ColorFormat::NV12_TWO_PLANES);
+    auto steps = PreProcessSteps();
+    steps.convert_color(ColorFormat::BGR);
+    f = PrePostProcessor()
+            .input(InputInfo().tensor(std::move(input_tensor_info)).preprocess(std::move(steps)))
+            .build(f);
+
+    EXPECT_EQ(f->get_parameters().size(), 2);
+    EXPECT_EQ(f->get_parameters()[0]->get_element_type(), element::u8);
+    EXPECT_EQ(f->get_parameters()[0]->get_partial_shape(), (PartialShape{1, 2, 2, 1}));
+    EXPECT_EQ(f->get_parameters()[1]->get_element_type(), element::u8);
+    EXPECT_EQ(f->get_parameters()[1]->get_partial_shape(), (PartialShape{1, 1, 1, 2}));
+}
+
+TEST(pre_post_process, convert_color_nv12_bgr_2_planes_el_type) {
+    auto f = create_simple_function(element::u8, Shape{1, 2, 2, 3});
+    EXPECT_NO_THROW(
+        f = PrePostProcessor()
+                .input(InputInfo()
+                           .tensor(InputTensorInfo()
+                                       .set_element_type(element::f32)
+                                       .set_color_format(ColorFormat::NV12_TWO_PLANES))
+                           .preprocess(
+                               PreProcessSteps().convert_element_type(element::u8).convert_color(ColorFormat::BGR)))
+                .build(f));
+
+    EXPECT_EQ(f->get_parameters().size(), 2);
+    EXPECT_EQ(f->get_parameters()[0]->get_element_type(), element::f32);
+    EXPECT_EQ(f->get_parameters()[1]->get_element_type(), element::f32);
+}
+
+TEST(pre_post_process, convert_color_unsupported) {
+    // Feel free to update this test when more color conversions are supported in future
+    auto f = create_simple_function(element::f32, PartialShape{1, 4, 4, 3});
+    EXPECT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_color_format(ColorFormat::NV12_SINGLE_PLANE))
+                                    .preprocess(PreProcessSteps().convert_color(ColorFormat::UNDEFINED)))
+                         .build(f),
+                 ov::AssertFailure);
+    EXPECT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_color_format(ColorFormat::NV12_TWO_PLANES))
```

Comment:
do we have tests for sub names?

Reply:
Yes, there are such tests. Added more to verify match of sub-names number and number of planes